src/gui/CMakeLists.txt

```diff
@@ -733,6 +733,17 @@ SET(QGIS_GUI_MOC_HDRS
   processing/qgsprocessingwidgetwrapper.h
   processing/qgsprocessingwidgetwrapperimpl.h
 )
+FIND_PACKAGE(Qt5Qml REQUIRED)
+FIND_PACKAGE(Qt5QuickWidgets REQUIRED)
```

Comment:
I would put this into root CMakeLists where all other libs are searched for

Reply:
This would also make this a requirement for server, no?

Reply:
Not if it is in conditioned by WITH_GUI (or WITH_DESKTOP)

Reply:
I'm rather -1 on this anyway, keeping things modularized helps to keep on overview of what's used where for future changes.

Reply:
I think at least for consistency it makes sense to have it in root cmakelists... and it does not require people to look through N cmakelists to figure out the dependencies...

Reply:
On the other hand, there is no need to keep conditionals (if with_gui or with_quick or with_whatevereelse) up to date if it's defined where it's actually required.
I don't think CMakeLists is the place where people (should) look for dependencies. There's the installation instruction, package sources or cmake output.

Reply:
There is already a WITH_GUI
https://github.com/qgis/QGIS/blob/master/CMakeLists.txt#L363-L365

Reply:
I'm not sure "keeping consistency" is really an argument here.

```
./src/auth/oauth2/CMakeLists.txt:  FIND_PACKAGE(QtKeychain REQUIRED)
./src/auth/oauth2/CMakeLists.txt:  FIND_PACKAGE(O2 REQUIRED)
./src/plugins/globe/CMakeLists.txt:FIND_PACKAGE(OSG REQUIRED)
./src/plugins/globe/CMakeLists.txt:FIND_PACKAGE(OSGEARTH REQUIRED)
./src/plugins/globe/CMakeLists.txt:FIND_PACKAGE(Qt5OpenGL REQUIRED)
./src/plugins/georeferencer/CMakeLists.txt:FIND_PACKAGE(GSL REQUIRED)      # Georeferencer
./src/native/CMakeLists.txt:FIND_PACKAGE(Qt5DBus REQUIRED)
./src/native/CMakeLists.txt:  FIND_PACKAGE(Qt5MacExtras)
./src/native/CMakeLists.txt:  FIND_PACKAGE(Qt5WinExtras)
./src/app/CMakeLists.txt:  FIND_PACKAGE(QwtPolar)
./src/app/CMakeLists.txt:      FIND_PROGRAM(WINDRES windres)
./src/server/CMakeLists.txt:FIND_PACKAGE(Fcgi REQUIRED)
./src/customwidgets/CMakeLists.txt:FIND_PACKAGE(Qt5UiPlugin REQUIRED)
./src/customwidgets/CMakeLists.txt:FIND_PACKAGE(Qt5Designer REQUIRED)
./src/providers/oracle/ocispatial/CMakeLists.txt:FIND_PACKAGE(OCI REQUIRED)
./src/providers/wms/CMakeLists.txt:FIND_PACKAGE(Qt5XmlPatterns REQUIRED)
./src/providers/mdal/CMakeLists.txt:FIND_PACKAGE(HDF5)
./src/providers/mdal/CMakeLists.txt:FIND_PACKAGE(NetCDF)
```